src/test/kotlin/html4tree/SymlinkReadTest.kt

```
@@ -0,0 +1,43 @@
+
+    @Test
+    fun testProcessIgnoreFileDoesNotFollowSymlink() {
+        val tempDir = Files.createTempDirectory("test-symlink-read-")
+        try {
+            val secretFile = tempDir.resolve("secret.txt")
+            Files.write(secretFile, listOf("top_secret_pattern"))
+
+            val symlinkFile = tempDir.resolve(".html4ignore")
+            try {
+                Files.createSymbolicLink(symlinkFile, secretFile)
+            } catch (e: UnsupportedOperationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)